SingletonApp/Classes/MyConnection.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.Data.Sqlite;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Data;
+
+namespace SingletonApp
+{
+    class MyConnection: IDisposable
+    {
+        private static MyConnection instance;        
+        private bool disposed = false;
+        private static SqliteConnection connection;
+        private MyConnection()
+        {
+            connection = new SqliteConnection(@"Data Source=mydb.db");
+            connection.Open();            
```

Comment:
Move this line to ExecuteQuery method to open the connection only when it's really needed